cmd/metal-api/internal/grpc/waitServer.go

```diff
@@ -159,14 +162,30 @@ func (s *WaitServer) Wait(req *v1.WaitRequest, srv v1.Wait_WaitServer) error {
 				}
 				nextCheck = now.Add(60 * time.Second)
 			}
-			err := srv.Send(&v1.WaitResponse{})
+			err := sendKeepPatientResponse(srv)
 			if err != nil {
 				return err
 			}
 		}
 	}
 }
 
+// https://github.com/grpc/grpc-go/issues/1229#issuecomment-302755717
+func sendKeepPatientResponse(srv v1.Wait_WaitServer) error {
+	errChan := make(chan error, 1)
+	ctx := srv.Context()
```

Comment:
Just for me to know... Under which circumstances does this context receive a `Done()`?

Reply:
Whenever the stream to the client completes or breaks

Reply:
Okay, then let's do it.